### ISSUE TITLE
[v6] Support union types on responses

### ISSUE THREAD

**Issue by joheredi** [Azure/autorest.typescript#585]
A response can specify an enum of expected values. For example

```json
 "get": {
                "operationId": "bool_getTrue",
                "description": "Get true Boolean value",
                "tags": [
                    "Bool Operations"
                ],
                "responses": {
                    "200": {
                        "description": "The true Boolean value",
                        "schema": {
                            "type": "boolean",
                            "enum": [true]
                        }
                    },
                    "default": {
                        "description": "Unexpected error",
                        "schema": {
                            "$ref": "#/definitions/Error"
                        }
                    }
                }
            }
```

Currently the Response type ignores the enum property and sets the type to the schema type. In the case above, body would be of type boolean.

Ideally we should support this and have body being a union type of the values specified in the enum property. For example:

```ts
type GetResponseBody = true;
```

Sample swaggers
https://github.com/Azure/autorest.testserver/blob/5b9079f374c0a95c1ff22aa5661ff90088304640/swagger/body-boolean.json#L27

https://github.com/Azure/autorest.testserver/blob/5b9079f374c0a95c1ff22aa5661ff90088304640/swagger/body-string.json#L128

**Comment by StupydHors** [comment]
So, how should this be handled properly? 
Even after 4 years, I cannot find a concrete example of this. Many of my endpoints can return 401 (Unauthorized), and it seems it is completely ignored.
